cli/pcluster/config/mappings.py

```diff
@@ -655,40 +656,246 @@
     ])
 }
 
-CLUSTER = {
+CLUSTER_COMMON_PARAMS = [
+    # OrderedDict due to conditional defaults values
+    ("cluster_config_metadata", {
+        # NOTE: this is not exposed as a configuration parameter
+        "type": ClusterConfigMetadataCfnParam,
+        "cfn_param_mapping": "ClusterConfigMetadata",
+        "update_policy": UpdatePolicy.IGNORED,
+        "visibility": Visibility.PRIVATE,
+    }),
+    ("key_name", {
```

Comment:
We can set it as required

Reply:
What about having a unit test for required parameters?

Reply:
Could we leave this outside the patch? This change has a big impact on tests and the issue was not introduced here.

Reply:
ok